### PR TITLE
Strip assigns from the email when testing

### PR DESCRIPTION
**Pull request by andrewtimberlake**
I needed this because I’m passing `conn` into some templates
Implements #154

### CODE REVIEW

File: test/lib/bamboo/adapters/test_adapter_test.exs [hunk 1]
```diff
@@ -157,4 +159,16 @@ defmodule Bamboo.TestAdapterTest do
 
     assert_delivered_email sent_email
   end
+
+  # The actual sent email and the email tested may have differences in their assigns
+  # Even if the resulting body is the same
+  test "assertion helpers remove assigns" do
```
Comment (paulcsmith) [inline test/lib/bamboo/adapters/test_adapter_test.exs]: I think the comment can be removed and maybe the test can say something like `"delivered emails have normalized assigns"` or something like that. What do you think?

File: test/lib/bamboo/adapters/test_adapter_test.exs [hunk 2]
```diff
@@ -157,4 +159,16 @@ defmodule Bamboo.TestAdapterTest do
...
+    test_email = new_email(from: "foo@bar.com", to: "bar@baz.com")
+    sent_email = %{test_email | assigns: %{foo: :bar}}
+    expected_email = %{test_email | assigns: %{foo: :baz}}
+
+    sent_email |> TestMailer.deliver_now
+
+    assert_delivered_email expected_email
```
Comment (paulcsmith) [inline test/lib/bamboo/adapters/test_adapter_test.exs]: I think this test could be simplified a bit by doing something like

```elixir
email = new_email(from: "foo@bar.com", to: "bar@baz.com", assigns: :anything)

email |> TestMailer.deliver_now

assert_delivered_email %{email | assigns: :assigns_removed_for_testing}
```

That way the assertion more clearly shows the expected result. What do you think?

**Comment by paulcsmith** [comment]
Two small comments in the test. The rest looks perfect. Thank you very much!

**Comment by andrewtimberlake** [comment]
Good comments, I’ve implemented both changes.

**Comment by paulcsmith** [comment]
Thank you so much :D

**Pull request merged**